### DISCUSSION TITLE
 ERR_PNPM_LINKING_FAILED  Error: EPERM: operation not permitted, rename

### DISCUSSION THREAD

**Discussion by mhamri** [pnpm/pnpm#7247] [Q&A]
to be frank I'm tired by number of time I'm receiving this error on daily basis. 

any suggestion how to deal with it? each time it's a random package name. if it's permission error, why everytime it's a different package? if it's an error, why it works when I run pnpm install for second or third time?

**Comment by sharky98** [comment]
Yeah, I have the same issues. I suspect your are on Windows; and there is some async stuff going on where something in pnpm is still working on the file, locking it; which prevent another part of pnpm to create the symlink (or whatever which version they are using).

**Reply by hrutik7** [reply]
yes I am on windows and for some reason I switched over wsl
and after returning back I am getting this error any update on this ?
